nessie/src/test/java/org/apache/iceberg/nessie/TestNessieTable.java

```diff
@@ -356,6 +360,107 @@ public void testListTables() {
     Assertions.assertThat(catalog.tableExists(TABLE_IDENTIFIER)).isTrue();
   }
 
+  @Test
+  public void testRegisterTableWithGivenBranch() {
+    List<String> metadataVersionFiles = metadataVersionFiles(TABLE_NAME);
+    Assertions.assertThat(1).isEqualTo(metadataVersionFiles.size());
+    catalog.dropTable(TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@main`"), false);
+    catalog.registerTable(
+            TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@main`"),
+            "file:" + metadataVersionFiles.get(0));
+    Table newTable = catalog.loadTable(TableIdentifier.of(DB_NAME, "`" + TABLE_NAME + "@main`"));
+    Assertions.assertThat(newTable).isNotNull();
```

Comment:
can you extract line 372 to 376 in a common function and reuse in other the testcases?